components/org.wso2.carbon.deployment.engine/src/main/java/org/wso2/carbon/deployment/engine/internal/DeploymentEngine.java

```diff
@@ -93,15 +98,22 @@ public DeploymentEngine() {
      * This will start the repository scanner and scheduler task and load artifacts to
      * the deployment engine.
      *
-     * @param repositoryDir the deployment repository directory that repository scanner will start scanning.
+     * @param serverRepositoryDir server deployment repository directory that repository scanner will start scanning.
+     * @param runtimeRepositoryDir runtime deployment repository directory that repository scanner will start scanning.
      * @throws DeploymentEngineException when an error occurs while trying to start the deployment engine.
      */
-    public void start(String repositoryDir) throws DeploymentEngineException {
-        logger.debug("Starting carbon deployment engine for repository : " + repositoryDir);
-        repositoryDirectory = new File(repositoryDir);
-        if (!repositoryDirectory.exists()) {
-            throw new DeploymentEngineException("Cannot find repository : " + repositoryDirectory);
+    public void start(String serverRepositoryDir, String runtimeRepositoryDir) throws DeploymentEngineException {
```

Comment:
What is the usage of String serverRepositoryDir, String runtimeRepositoryDir here ? We are just validating their presence, but not using anywhere.